src/polyswarmclient/abstractscanner.py

```diff
@@ -46,15 +100,14 @@ async def scan(self, guid, artifact_type, content, metadata, chain):
         Returns:
             ScanResult: Result of this scan
         """
-        pass
+        raise NotImplementedError("Must implement scan_async when using ScanMode.Async")
 
     async def setup(self):
         """Override this method to implement custom setup logic.
 
         This is run immediately after the Scanner class is instantiated and before any calls to the scan() method.
 
         Args:
```

Comment:
Consider to remove docstrings on `Args:` if no arg will be described.
```suggestion
        This is run immediately after the Scanner class is instantiated and before any calls to the scan() method.
```